packages/editor/src/lib/app/statechart/TLSelectTool/children/Idle.ts

```diff
@@ -267,43 +267,71 @@ export class Idle extends StateNode {
 	}
 
 	onKeyUp = (info: TLKeyboardEventInfo) => {
-		if (this.app.isReadOnly) return
-		switch (info.code) {
-			case 'Enter': {
-				const { selectedShapes } = this.app
-
-				if (selectedShapes.every((shape) => shape.type === 'group')) {
-					this.app.setSelectedIds(
-						selectedShapes.flatMap((shape) => this.app.getSortedChildIds(shape.id))
-					)
-					return
+		if (this.app.isReadOnly) {
```

Comment:
Felt better to create a separate branch for readonly and only handle the single scenario that applies to that mode.